incubator/hackmd/templates/deployment.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "hackmd.fullname" . }}
+  labels:
+    app: {{ template "hackmd.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "hackmd.name" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - containerPort: {{ .Values.service.internalPort }}
+          args:
+            - "/bin/bash"
+            - "-c"
+            - "dockerize -wait tcp://{{ template "postgresql.fullname" . }}:5432 -timeout 30s /hackmd/docker-entrypoint.sh"
+          livenessProbe:
+            httpGet:
+              path: /status
+              port: {{ .Values.service.internalPort }}
+            initialDelaySeconds: 120
+          readinessProbe:
+            httpGet:
+              path: /status
+              port: {{ .Values.service.internalPort }}
+            initialDelaySeconds: 30
+          env:
+            - name: HMD_DB_URL
+              value: postgres://{{ .Values.postgresql.postgresUser }}:{{ .Values.postgresql.postgresPassword }}@{{ template "postgresql.fullname" . }}:5432/{{ .Values.postgresql.postgresDatabase }}
+            {{- range $key, $value :=  .Values.env }}
+            - name: {{ $key | upper | replace "-" "_" }}
+              value: {{ $value | quote }}
+            {{- end }}
+          volumeMounts:
+            - name: config
+              subPath: config.json
+              mountPath: /hackmd/config.json
```

Comment:
You actually don't need this if you use environment variables to configure HackMD.

https://github.com/hackmdio/hackmd#environment-variables-will-overwrite-other-server-configs